tests/test_forms.py

```diff
@@ -141,3 +141,20 @@ def test(self, app, required_field_email, optional_field_email, unspecified_fiel
 
             if unspecified_field_email in (self._invalid_address, ""):
                 assert form.errors["unspecified_email"] == ["Please enter a valid email address."]
+
+
+class TestRemoveCsrfToken:
+    def test_csrf_token_removed_if_present(self):
+        data = {'key': 'value', 'key2': 'value2', 'csrf_token': '123tgredsca2345ywt34rwe'}
+        cleaned_data = remove_csrf_token(data)
+        assert 'csrf_token' not in cleaned_data
+
+    def test_original_data_not_mutated_by_remove_csrf_token(self):
```

Comment:
To check the data isn't mutated, you could test the value of the csrf key?

Reply:
Shouldn't the csrf key _not_ be in the data? Should you assert that the other keys are still there?

Reply:
By mutated I mean that the original dict shouldn't be changed - i.e. the method should make a copy of the data and then remove `csrf_token` only from the copy, leaving the original as-is.

This test checks that the original dict (`data`) still contains `csrf_token` after processing.